### PR TITLE
adding defaults to json files

### PR DESCRIPTION
**Pull request by plgbrts**
In this PR some defaults have been aded to the json files `COMPTRAJ` and `WELTRAJ`.

Furthermore `WELSPECS` json file is adjusted to allow defaults (`1*`) for `HEAD_I` and `HEAD_J`. With this change  of `WELSPECS` the (standard) well can fully be specified grid independent together with `WELTRAJ` and `COMPTRAJ` keywords.

### CODE REVIEW

**Comment by plgbrts** [comment]
jenkins build this please

**Comment by plgbrts** [comment]
The effect of using defaults in `WELSPECS` appears in visualization only. The red part (the relevant part for simulation) , the part open to flow, has the same grids intersected and same well connection factors.
For the left figure 
```
WELSPECS
-- Item #: 1	 2	3	4	5	 6
	'INJ'	'G1'	3	3	8335	'GAS' /
/
```
is used and for the right figure 
```
WELSPECS
-- Item #: 1	 2	3	4	5	 6
	'INJ'	'G1'	1*	1*	8335	'GAS' /
/
```

image

**Review by bska — COMMENTED** [review]
Adding the defaults to our 900_OPM keywords is a good thing, but I think we need to do more work if we want to fully support defaulting `HEAD_I` and `HEAD_J` in `WELSPECS`.  At minimum, there should be a way to get reasonable values for these&mdash;e.g., the (I,J) coordinates of the top-most reservoir connection&mdash;when calling [`Well::getHeadI()`](https://github.com/OPM/opm-common/blob/24d25f7be7a7e3fd62591e668b1612051d99fb20/src/opm/input/eclipse/Schedule/Well/Well.cpp#L969-L971) and [`Well::getHeadJ()`](https://github.com/OPM/opm-common/blob/24d25f7be7a7e3fd62591e668b1612051d99fb20/src/opm/input/eclipse/Schedule/Well/Well.cpp#L973-L975).  If we don't have reasonable values here, the plots will look almost arbitrarily poor in a post-processing tool such as ResInsight.

On a tangential note, there should probably also be a way to reject `COMPORD` as anything other than `INPUT` for the wells defined using `WELTRAJ`/`COMPTRAJ`.

**Comment by plgbrts** [comment]
jenkins build this please

**Comment by plgbrts** [comment]
With defaulted `Head_I `& `Head_J` in WELSPECS the `Head_I` and ` Head_J ` is now calculated based on the well trajectory data.  Wells in ResInsight should now look descent using defaults.

Example:

```
     WELSPECS
     'INJ'	'G1'	1*	1*	8335	'GAS' /
     /

     WELTRAJ
     -- WELL           BR_NO       X          Y              TVD         MD 
     'INJ'             1*          2500       500           -100.0	0.0 /
     'INJ'             1*          2500       500            8325.0	8325.0/
     'INJ'             1*          4500       500            8425.0	8450.0 /
     /

     COMPTRAJ 
     -- WELL	 BRANCH  PERF	  PERF    PERF COMPL STATE SAT    CONN DIAM	KH	SKIN D_FACT 
     -- NAME  NO      TOP      BOT     REF  NO    --    TABLE  FACT --   --  --   --     
     'INJ'	 1* 	 8325.0   8450.0  1*   1*	 1*    1*     1*   0.5  1*  0.1  1*  /
     /
```
Corresponding Plot (`DX=DY=1000 ft`):
image

I have not addressed (it is not clear to me yet how to do this) the rejecting of certain `COMPORD` options 
and propose a separate PR for this.

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -220,6 +220,11 @@ namespace {
                     this->snapshots.back().wells.update( well2 );
                 }
                 this->snapshots.back().wellgroup_events().addEvent( name, ScheduleEvents::COMPLETION_CHANGE);
+                const auto& md = connections->getMD();
+                if (!std::is_sorted(std::begin(md), std::end(md))) {
+                    auto msg = fmt::format("Well {} has no increasing measured depth column", name);
```
Comment (goncalvesmachadoc) [inline src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:225]: Wouldn't be clearer to say something like: Well {} measured depth column in the WELLTRAJ keyword is not strictly increasing.

Comment (plgbrts) [reply src/opm/input/eclipse/Schedule/KeywordHandlers.cpp]: That is better. I changed the error message. It gives now

```
Error: Problem with keyword WELTRAJ
In SPE1CASE1_TRAJ_WELSPEC_DEF.DATA line 410
Well INJ measured depth column is not strictly increasing

Error: Unrecoverable errors while loading input: Problem with keyword WELTRAJ
In SPE1CASE1_TRAJ_WELSPEC_DEF.DATA line 410
Well INJ measured depth column is not strictly increasing
```
The error message is repeated for some reason but I believe that this issue is addressed in https://github.com/OPM/opm-common/pull/3524

**Comment by plgbrts** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
This looks good to me now.  The main ability of inferring the intersection between the reservoir and the well from its top-most reservoir connection is very useful.  I just noticed one potential inefficiency in the return type of `WellConnections::getMD()`.  If you need to return a full vector then that's fine, otherwise we should consider returning a "reference-to-`const`" instead.

**Comment by plgbrts** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**